src/lib.rs

```diff
@@ -251,6 +251,15 @@ mod tests {
         assert!(m != "bar");
     }
 
+    #[test]
+    fn compare_wild_match() {
+        assert_eq!(WildMatch::default(), WildMatch::new(""));
+        assert_eq!(WildMatch::new("abc"), WildMatch::new("abc"));
+        assert_eq!(WildMatch::new("a*bc"), WildMatch::new("a*bc"));
```

Comment:
Would `assert_eq!(WildMatch::new("a***c"), WildMatch::new("a*c"));` also work? Because the semantic of both is the same.

Reply:
I wouldn't have thought that it worked but it did because you simplify the pattern already at construction time.